Signal/src/ViewControllers/NotificationSettingsViewController.m

```diff
@@ -38,21 +38,15 @@ - (void)updateTableContents
     OWSTableSection *soundsSection = [OWSTableSection new];
     soundsSection.headerTitle
         = NSLocalizedString(@"SETTINGS_SECTION_SOUNDS", @"Header Label for the sounds section of settings views.");
-
-    [soundsSection addItem:[OWSTableItem itemWithCustomCellBlock:^{
-        UITableViewCell *cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue1
-                                                       reuseIdentifier:@"UITableViewCellStyleValue1"];
-        cell.textLabel.text = NSLocalizedString(@"SETTINGS_ITEM_NOTIFICATION_SOUND",
-            @"Label for settings view that allows user to change the notification sound.");
-        OWSSound sound = [OWSSounds globalNotificationSound];
-        cell.detailTextLabel.text = [OWSSounds displayNameForSound:sound];
-        [cell setAccessoryType:UITableViewCellAccessoryDisclosureIndicator];
-        return cell;
-    }
-                               actionBlock:^{
-                                   OWSSoundSettingsViewController *vc = [OWSSoundSettingsViewController new];
-                                   [weakSelf.navigationController pushViewController:vc animated:YES];
-                               }]];
+    [soundsSection
+        addItem:[OWSTableItem disclosureItemWithText:
```

Comment:
I DRY'd up some code by adding a new `[OWSTableItem disclosureItemWithText: detailText: actionBlock:]` method for the case where you want "detail text".

Reply:
👍